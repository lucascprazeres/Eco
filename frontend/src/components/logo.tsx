import { Box, Typography } from '@mui/material'
import Image from 'next/image'

import logoImg from '@eco/assets/logo.svg'

export function Logo() {
  return (
    <Box display="flex" alignItems="center" gap={1}>
      <Typography
        variant="h4"
        color="primary.dark"
        fontFamily="ubuntu"
        fontWeight="700"
      >
        Eco
      </Typography>
      <Image src={logoImg} alt="" height={40} width={40} />
    </Box>
  )
}
