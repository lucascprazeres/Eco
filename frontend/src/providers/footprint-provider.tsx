import { useMutation } from '@apollo/client'
import { CALCULATE_CARBON_FOOTPRINT } from '@eco/graphql/mutation'
import {
  CalculateFootprintMutationOutput,
  CarbonFootprint,
  CarbonFootprintInput,
} from '@eco/models/carbon-footprint'
import {
  createContext,
  PropsWithChildren,
  useCallback,
  useContext,
  useState,
} from 'react'

interface FootprintContextData {
  footprint: CarbonFootprint | null
  handleCalculateFootprint: (data: CarbonFootprintInput) => Promise<void>
}

const FootprintContext = createContext<FootprintContextData>(
  {} as FootprintContextData,
)

export function FootprintProvider({ children }: PropsWithChildren) {
  const [footprint, setFootprint] = useState<CarbonFootprint | null>(null)

  const [calculateFootprint] = useMutation<CalculateFootprintMutationOutput>(
    CALCULATE_CARBON_FOOTPRINT,
  )

  const handleCalculateFootprint = useCallback(
    async (formData: CarbonFootprintInput) => {
      const { data } = await calculateFootprint({
        variables: {
          input: formData,
        },
      })

      setFootprint(data?.calculateFootprint as unknown as CarbonFootprint)
    },
    [calculateFootprint],
  )

  return (
    <FootprintContext.Provider value={{ footprint, handleCalculateFootprint }}>
      {children}
    </FootprintContext.Provider>
  )
}

export const useFootprint = () => useContext(FootprintContext)
