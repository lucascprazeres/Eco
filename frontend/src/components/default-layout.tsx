import { Box } from '@mui/material'
import { ReactElement } from 'react'
import { Logo } from './logo'

interface DefaultLayoutProps {
  children: ReactElement | ReactElement[]
}

export function DefaultLayout({ children }: DefaultLayoutProps) {
  return (
    <Box height="100vh" width="100vw">
      <Box component="header" width="100%" padding={2} height={72}>
        <Logo />
      </Box>

      <Box
        height="calc(100% - 72px)"
        component="main"
        display="flex"
        flexDirection="column"
        alignItems="center"
        justifyContent="center"
      >
        {children}
      </Box>
    </Box>
  )
}
