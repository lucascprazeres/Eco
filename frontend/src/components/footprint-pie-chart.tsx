import { useFootprint } from '@eco/providers/footprint-provider'
import { toPercentage } from '@eco/utils/percentage'
import { PieChart } from '@mui/x-charts'
import { useMemo } from 'react'

export function FootprintPieChart() {
  const { footprint } = useFootprint()

  const pieChartData = useMemo(() => {
    if (!footprint) return []

    return [
      {
        label: 'Electricity Usage (%)',
        value: toPercentage(footprint.electricity.value, footprint.total.value),
      },
      {
        label: 'Transportation fuel (%)',
        value: toPercentage(
          footprint.transportation.value,
          footprint.total.value,
        ),
      },
      {
        label: 'Air travel (%)',
        value: toPercentage(footprint.airTravel.value, footprint.total.value),
      },
    ]
  }, [footprint])

  return (
    <PieChart
      series={[
        {
          data: pieChartData,
          cx: 0,
          highlightScope: { faded: 'global', highlighted: 'item' },
          faded: {
            innerRadius: 20,
            additionalRadius: -20,
            color: 'gray',
          },
        },
      ]}
      height={350}
      width={300}
      colors={['#34CB79', '#322153', '#8257E5']}
      slotProps={{ legend: { padding: -40 } }}
    />
  )
}
