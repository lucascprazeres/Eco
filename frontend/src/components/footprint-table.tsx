import { useFootprint } from '@eco/providers/footprint-provider'
import {
  Paper,
  styled,
  Table,
  TableBody,
  TableCell,
  tableCellClasses,
  TableContainer,
  TableHead,
  TableRow,
} from '@mui/material'
import { useMemo } from 'react'

const StyledTableCell = styled(TableCell)(({ theme }) => ({
  [`&.${tableCellClasses.head}`]: {
    backgroundColor: theme.palette.background.default,
    fontSize: 16,
    fontWeight: '700',
  },
  [`&.${tableCellClasses.body}`]: {
    fontSize: 15,
  },
}))

export function FootprintTable() {
  const { footprint } = useFootprint()

  const footprintBreakdownRows = useMemo(() => {
    if (!footprint) return []

    return [
      {
        name: 'Electricity Usage  (Kwh/year)',
        emission: `${footprint?.electricity?.value} ${footprint?.electricity?.unit}`,
      },
      {
        name: 'Transportation fuel  (Gallons/year)',
        emission: `${footprint?.transportation?.value} ${footprint?.transportation?.unit}`,
      },
      {
        name: 'Air travel  (travels/year)',
        emission: `${footprint?.airTravel?.value} ${footprint?.airTravel?.unit}`,
      },
      {
        name: 'Total',
        emission: `${footprint?.total?.value} ${footprint?.total?.unit}`,
      },
    ]
  }, [footprint])

  return (
    <TableContainer variant="outlined" component={Paper} sx={{ width: 600 }}>
      <Table>
        <TableHead>
          <TableRow>
            <StyledTableCell>Name</StyledTableCell>
            <StyledTableCell>Emission</StyledTableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {footprintBreakdownRows.map((row) => (
            <TableRow
              key={row.name}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <StyledTableCell>{row.name}</StyledTableCell>
              <StyledTableCell sx={{ fontWeight: 'bold' }}>
                {row.emission}
              </StyledTableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  )
}
