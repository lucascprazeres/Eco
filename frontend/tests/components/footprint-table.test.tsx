import { describe, it, expect, vi, afterEach, Mock } from 'vitest'
import { render, screen, cleanup } from '@testing-library/react'
import { FootprintTable } from '@eco/components/footprint-table'
import { useFootprint } from '@eco/providers/footprint-provider'

vi.mock('@eco/providers/footprint-provider')

describe('FootprintTable', () => {
  afterEach(() => {
    cleanup()
    vi.clearAllMocks()
  })

  it('should be able to render correctly', () => {
    // arrange
    // eslint-disable-next-line prettier/prettier
    (useFootprint as Mock).mockReturnValueOnce({
      footprint: {
        electricity: { value: 1200, unit: 'kgCO2e/year' },
        transportation: { value: 150, unit: 'kgCO2e/year' },
        airTravel: { value: 100, unit: 'kgCO2e/year' },
        total: { value: 2000, unit: 'kgCO2e/year' },
      },
    })

    // act
    const result = render(<FootprintTable />)

    // assert
    expect(screen.getByText('Name')).toBeInTheDocument()
    expect(screen.getByText('Emission')).toBeInTheDocument()

    expect(screen.getByText('Electricity Usage (Kwh/year)')).toBeInTheDocument()
    expect(screen.getByText('1200 kgCO2e/year')).toBeInTheDocument()

    expect(
      screen.getByText('Transportation fuel (Gallons/year)'),
    ).toBeInTheDocument()
    expect(screen.getByText('150 kgCO2e/year')).toBeInTheDocument()

    expect(screen.getByText('Air travel (travels/year)')).toBeInTheDocument()
    expect(screen.getByText('100 kgCO2e/year')).toBeInTheDocument()

    expect(screen.getByText('Total')).toBeInTheDocument()
    expect(screen.getByText('2000 kgCO2e/year')).toBeInTheDocument()

    expect(result).toMatchSnapshot()
  })

  it('should be able to handle an empty footprint', () => {
    // arrange
    // eslint-disable-next-line prettier/prettier
    (useFootprint as Mock).mockReturnValue({ footprint: null })

    // act
    render(<FootprintTable />)

    // assert
    expect(
      screen.queryByText('Electricity Usage  (Kwh/year)'),
    ).not.toBeInTheDocument()
    expect(
      screen.queryByText('Transportation fuel  (Gallons/year)'),
    ).not.toBeInTheDocument()
    expect(
      screen.queryByText('Air travel  (travels/year)'),
    ).not.toBeInTheDocument()
    expect(screen.queryByText('Total')).not.toBeInTheDocument()
  })
})
