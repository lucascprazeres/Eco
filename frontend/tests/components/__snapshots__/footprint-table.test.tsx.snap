// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FootprintTable > should be able to render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-102pnb2-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
                scope="col"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
                scope="col"
              >
                Emission
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
              >
                Electricity Usage  (Kwh/year)
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
              >
                1200 kgCO2e/year
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
              >
                Transportation fuel  (Gallons/year)
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
              >
                150 kgCO2e/year
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
              >
                Air travel  (travels/year)
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
              >
                100 kgCO2e/year
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
              >
                Total
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
              >
                2000 kgCO2e/year
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-102pnb2-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-1xwxv7r-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
              scope="col"
            >
              Emission
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
            >
              Electricity Usage  (Kwh/year)
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
            >
              1200 kgCO2e/year
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
            >
              Transportation fuel  (Gallons/year)
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
            >
              150 kgCO2e/year
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
            >
              Air travel  (travels/year)
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
            >
              100 kgCO2e/year
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
            >
              Total
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
            >
              2000 kgCO2e/year
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FootprintTable > should be able to render the table headers correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-102pnb2-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
                scope="col"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
                scope="col"
              >
                Emission
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-102pnb2-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-1xwxv7r-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
              scope="col"
            >
              Emission
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
        />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FootprintTable > should render rows based on the footprint data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-102pnb2-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
                scope="col"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
                scope="col"
              >
                Emission
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
              >
                Electricity Usage  (Kwh/year)
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
              >
                1200 kgCO2e/year
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
              >
                Transportation fuel  (Gallons/year)
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
              >
                150 kgCO2e/year
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
              >
                Air travel  (travels/year)
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
              >
                100 kgCO2e/year
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
              >
                Total
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
              >
                2000 kgCO2e/year
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-102pnb2-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-1xwxv7r-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
              scope="col"
            >
              Emission
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
            >
              Electricity Usage  (Kwh/year)
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
            >
              1200 kgCO2e/year
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
            >
              Transportation fuel  (Gallons/year)
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
            >
              150 kgCO2e/year
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
            >
              Air travel  (travels/year)
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
            >
              100 kgCO2e/year
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-jbs7qj-MuiTableCell-root"
            >
              Total
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1kq0cpe-MuiTableCell-root"
            >
              2000 kgCO2e/year
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FootprintTable > should render the table headers correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-102pnb2-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
                scope="col"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
                scope="col"
              >
                Emission
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiTableContainer-root css-102pnb2-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-1xwxv7r-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h02pjl-MuiTableCell-root"
              scope="col"
            >
              Emission
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
        />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
